src/test/scala/gremlin/scala/LogicalSpec.scala

```diff
@@ -0,0 +1,76 @@
+package gremlin.scala
+
+class LogicalSpec extends TestBase {
+
+  describe("and steps") {
+
+    it("returns a vertex if both conditions are met") {
+      val x = v(1)
+
+      x.and(
```

Comment:
Thinking more about this ... I guess syntax we want to rather achieve is:

```scala
x.and(
  _.out().has("name", "lop"),
  _.out().has("name", "vadas")
).values[String]("name").toSet should be(Set("marko"))
```

right? Would `and` annotation be `def and(traversals: (End ⇒ GremlinScala[End, HNil])*)`?